app/scripts/lib/router.js

```diff
@@ -159,17 +159,43 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Redirect the user to the best suitable OAuth flow
+     * Redirect the user to the best suitable OAuth flow.
      */
     redirectToBestOAuthChoice: function () {
-      var account = this.user.getChooserAccount();
+      var self = this;
       var route = '/oauth/signin';
 
-      if (account.isDefault()) {
-        route = '/oauth/signup';
-      }
+      var account = this.user.getChooserAccount();
 
-      return this.navigate(route, { replace: true, trigger: true });
+      // Attempt to get email address from relier
+      var email = this.broker.relier.get('email');
+
+      // Check whether email address corresponds to an existing account and
+      // navigate to correct login/signup page.
+      if (email) {
+        return account.checkAccountExistsByEmail(email)
+        .then(function (exists) {
+          if (! exists) {
+            route = '/oauth/signup';
+          }
+        }, function (err) {
+          // In case of error, choose navigation page based on
+          // whether account is a default account
+          if (account.isDefault()) {
```

Comment:
I should have been more description 🙁. The error in this scenario, would be a throttling error. If it occured, I wanted `redirectToBestOAuthChoice` to fallback to its original redirect option, which is based on whether the account is a default account. I am not sure we should notify the user of this error because it doesn't effect the ability to log into their account. What are your thoughts?

I will update description and have it log error.